cdap-app-fabric/src/main/java/co/cask/cdap/metadata/MetadataHttpHandler.java

```diff
@@ -80,737 +76,201 @@
-  public void getArtifactMetadata(HttpRequest request, HttpResponder responder,
-                                  @PathParam("namespace-id") String namespaceId,
-                                  @PathParam("artifact-name") String artifactName,
-                                  @PathParam("artifact-version") String artifactVersionStr,
-                                  @QueryParam("scope") String scope) throws NotFoundException, BadRequestException {
-    ArtifactId artifactId = new ArtifactId(namespaceId, artifactName, artifactVersionStr);
-    responder.sendJson(HttpResponseStatus.OK, GSON.toJson(getMetadata(artifactId, scope), SET_METADATA_RECORD_TYPE));
-  }
-
-  @GET
-  @Path("/namespaces/{namespace-id}/datasets/{dataset-id}/metadata")
-  public void getDatasetMetadata(HttpRequest request, HttpResponder responder,
-                                 @PathParam("namespace-id") String namespaceId,
-                                 @PathParam("dataset-id") String datasetId,
-                                 @QueryParam("scope") String scope) throws NotFoundException, BadRequestException {
-    DatasetId datasetInstance = new DatasetId(namespaceId, datasetId);
-    responder.sendJson(HttpResponseStatus.OK,
-                       GSON.toJson(getMetadata(datasetInstance, scope), SET_METADATA_RECORD_TYPE));
-  }
-
-  @GET
-  @Path("/namespaces/{namespace-id}/streams/{stream-id}/metadata")
-  public void getStreamMetadata(HttpRequest request, HttpResponder responder,
-                                @PathParam("namespace-id") String namespaceId,
-                                @PathParam("stream-id") String streamId,
-                                @QueryParam("scope") String scope) throws NotFoundException, BadRequestException {
-    StreamId stream = new StreamId(namespaceId, streamId);
-    responder.sendJson(HttpResponseStatus.OK, GSON.toJson(getMetadata(stream, scope), SET_METADATA_RECORD_TYPE));
-  }
+  private static final Map<String, String> API_PART_TO_ENTITY_TYPE;
 
-  @GET
-  @Path("/namespaces/{namespace-id}/streams/{stream-id}/views/{view-id}/metadata")
-  public void getViewMetadata(HttpRequest request, HttpResponder responder,
-                              @PathParam("namespace-id") String namespaceId,
-                              @PathParam("stream-id") String streamId,
-                              @PathParam("view-id") String viewId,
-                              @QueryParam("scope") String scope) throws NotFoundException, BadRequestException {
-    StreamViewId view = new StreamViewId(namespaceId, streamId, viewId);
-    responder.sendJson(HttpResponseStatus.OK, GSON.toJson(getMetadata(view, scope), SET_METADATA_RECORD_TYPE));
+  // TODO(Rohit): Is there a better way to standardize this conversion of api parts to entities. Why do we have so many
+  // different representation? ex EntityTypeSimpleName. Investigate and refactor.
```

Comment:
I guess that is because these are plural? Are they always the same as the entity type name + "s"?